test/upgrade/installation/serverless.go

```diff
@@ -42,5 +42,13 @@ func UpgradeServerless(ctx *test.Context) error {
 		ctx.T.Error("Eventing upgrade failed:", err)
 	}
 
+	if _, err := v1a1test.WaitForKnativeKafkaState(ctx,
+		"knative-kafka",
+		knativeEventing,
+		v1a1test.IsKnativeKafkaReady,
```

Comment:
Do we need to validate the version here as well, like we do in Serving and Eventing?

Reply:
I think we do. The SRVKE-935 is there just because it was still running on the previous version of the KnativeKafka (0.23, instead of 0.24)

Reply:
The KnativeKafka CR's status doesn't have "version" field :-/

Reply:
Trying to see if the KnativeKafka resource was not Ready in those failed runs which would indicate that waiting only for "Ready" would be enough. But I don't see KnativeKafka CR in must-gather logs. Maybe the must-gather image doesn't collect KnativeKafka CR. Will check.
If we want to wait for the version as well we need to add the version handling to the operator.

Reply:
Sent https://github.com/openshift-knative/must-gather/pull/5 to fix must-gather

Reply:
There is also an alternative to use here. Just scrape the related metric [here](https://github.com/openshift-knative/serverless-operator/blob/3c0fc76d0c2e68a3b91e27075c9517256df6f22a/knative-operator/pkg/controller/knativekafka/knativekafka_controller.go#L152). We already use this method here: https://github.com/openshift-knative/serverless-operator/blob/3c0fc76d0c2e68a3b91e27075c9517256df6f22a/test/e2ekafka/knative_kafka_test.go#L97